### PR TITLE
disttask: persist business error to subtask table

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

- `WaitGlobalTask` may exit before cleaning up the global task, which can cause unexpected behavior.
- The error returned by `scheduler.Init()` should persist to subtask table instead of constantly retrying.

### What is changed and how it works?

As the title said.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/scheduler/manager.go [hunk 1]
```diff
@@ -373,11 +374,14 @@ func (m *Manager) removeHandlingTask(id int64) {
...
-	if err == nil {
-		return
-	}
-
+func (m *Manager) logErr(err error) {
 	logutil.Logger(m.logCtx).Error("task manager error", zap.Error(err))
 }
+
+func (m *Manager) logErrAndPersist(err error, taskID int64) {
+	m.logErr(err)
```
Comment (Benjamin2037) [inline disttask/framework/scheduler/manager.go:382]: Will here get return value of m.logErr(err), then store return packed err into subtask?

File: disttask/framework/scheduler/manager.go [hunk 2]
```diff
@@ -373,11 +374,14 @@ func (m *Manager) removeHandlingTask(id int64) {
...
 }
 
-// onError handles the error.
-func (m *Manager) onError(err error) {
-	if err == nil {
-		return
-	}
-
+func (m *Manager) logErr(err error) {
 	logutil.Logger(m.logCtx).Error("task manager error", zap.Error(err))
```
Comment (ywqzzy) [inline disttask/framework/scheduler/manager.go:378]: Add trace？

File: disttask/framework/scheduler/manager.go [hunk 3]
```diff
@@ -373,11 +374,14 @@ func (m *Manager) removeHandlingTask(id int64) {
...
 }
+
+func (m *Manager) logErrAndPersist(err error, taskID int64) {
+	m.logErr(err)
+	err1 := m.taskTable.UpdateErrorToSubtask(m.id, taskID, err)
```
Comment (ywqzzy) [inline disttask/framework/scheduler/manager.go:383]: When update to subtask failed, what will happen?
We can use backoffer.

File: disttask/framework/scheduler/manager.go (hunk 1 above)
Comment (tangenta) [reply disttask/framework/scheduler/manager.go:382]: It is designed NOT to store errors in subtasks. That's why I change into two methods.

File: disttask/framework/scheduler/manager.go (hunk 3 above)
Comment (tangenta) [reply disttask/framework/scheduler/manager.go:383]: It will act like `logErr()` and keep retrying.

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**